hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ResourceAttributeEnricher.java

```diff
@@ -77,4 +97,15 @@ private boolean isValidEvent(Event event) {
         && (event.getAttributes() != null)
         && (event.getAttributes().getAttributeMap() != null);
   }
+
+  private String getDeploymentType(String hostName) {
+    if (hostName.contains(EnrichedSpanConstants.getValue(Deployment.DEPLOYMENT_CANARY))) {
```

Comment:
use switch case no?

Reply:
it is not possible to use switch with contains. But I have removed if blocks by iterating over enum values instead

Reply:
Oh sorry missed the contains method. The new change is fine.